src/uct/cuda/base/cuda_md.c

```diff
@@ -55,6 +56,62 @@ UCS_PROFILE_FUNC(ucs_status_t, uct_cuda_base_detect_memory_type,
     return UCS_ERR_INVALID_ADDR;
 }
 
+static ucs_status_t uct_cuda_base_get_sys_dev(ucs_sys_device_t *sys_dev_p)
+{
+    CUdevice dev;
+    int attrib;
+
+    if (UCS_OK != UCT_CUDADRV_FUNC(cuCtxGetDevice(&dev))) {
+        return UCS_ERR_IO_ERROR;
+    }
+
+    if (UCS_OK != UCT_CUDADRV_FUNC(cuDeviceGetAttribute(&attrib,
+                    CU_DEVICE_ATTRIBUTE_PCI_DOMAIN_ID, dev))) {
+        return UCS_ERR_IO_ERROR;
+    }
+
+    sys_dev_p->bus_id.domain = (uint16_t)attrib;
+
+    if (UCS_OK != UCT_CUDADRV_FUNC(cuDeviceGetAttribute(&attrib,
+                    CU_DEVICE_ATTRIBUTE_PCI_BUS_ID, dev))) {
+        return UCS_ERR_IO_ERROR;
+    }
+
+    sys_dev_p->bus_id.bus = (uint8_t)attrib;
+
+    return UCS_OK;
+}
+
+ucs_status_t uct_cuda_base_mem_query(uct_md_h md, const void *addr,
```

Comment:
I think it should be:
```
ucs_status_t uct_cuda_base_mem_query(uct_md_h md, const void *addr,
                                     size_t length,
                                     uct_md_mem_attr_t *mem_attr_p)
{
    uint64_t filed_masks = mem_attr_p->filed_masks ;
    ucs_status_t status;

    mem_attr_p->field_mask = 0;

    if ((addr == NULL) || (length == 0)) {
        return UCS_ERR_INVALID_PARAM;
    }

    if (field_mask & UCT_MD_MEM_ATTR_FIELD_MEM_TYPE) {
        status = uct_cuda_base_detect_memory_type(md, addr, length,
                                                  &mem_attr_p->mem_type);
        if (UCS_OK != status) {
            ucs_warn("unable to query mem_type for %p", (void*) addr);
            return status;
        }

        mem_attr_p->field_mask |= UCT_MD_MEM_ATTR_FIELD_MEM_TYPE;
    }

    if (field_mask & UCT_MD_MEM_ATTR_FIELD_SYS_DEV) {
        status = uct_cuda_base_get_sys_dev(&mem_attr_p->sys_dev);
        if (UCS_OK != status) {
            ucs_warn("unable to query sys_dev for %p", (void*) addr);
            return status;
        }

        mem_attr_p->field_mask |= UCT_MD_MEM_ATTR_FIELD_SYS_DEV;
    }

    return UCS_OK;
}
```

So, user gets `MEM_TYPE | SYS_DEV` if they was requested. wdyt?

Reply:
@dmitrygx The semantics of this call changed based on the discussion with @yosefe here https://github.com/openucx/ucx/pull/4974#discussion_r408745110
Now users sets the fields that is of interest and the call tries to populate all requested fields. If any fail, then the function returns an error.

Reply:
> @dmitrygx The semantics of this call changed based on the discussion with @yosefe here [#4974 (comment)](https://github.com/openucx/ucx/pull/4974#discussion_r408745110)
> Now users set the fields that is of interest and the call tries to populate all requested fields. If any fail, then the function returns an error.

@Akshay-Venkatesh thank you! I see the idea now